src/Service.Host/client/src/components/PurchaseOrders/CreatePurchaseOrderUt.js

```diff
@@ -161,9 +164,20 @@ function CreatePurchaseOrderUt() {
         id: c.partNumber,
         name: c.partNumber,
         partNumber: c.partNumber,
-        description: c.description
+        description: c.description,
+        stockControlled: c.stockControlled
     }));
 
+    const partSuppliersSearchResults = useSelector(reduxState =>
+        collectionSelectorHelpers.getSearchItems(reduxState.partSuppliers)
+    );
+
+    useEffect(() => {
+        if (partSuppliersSearchResults?.length) {
+            handleSupplierChange(partSuppliersSearchResults[0]);
```

Comment:
I'd fish out the preferred supplier here as opposed to just choosing the first one

Reply:
Mm this was the original purpose of the sort that was implemented originally on the declaration of partSuppliersSearchResults (now just in the FetchItems). How else would I tell the preferred supplier other than choosing the first item of the sorted list (by supplier ranking)?

Reply:
partSupplierSearchResults won't be sorted by supplierRanking now since we moved the sort into the Dropdowns items list.

I'd just use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find